src/spyglass/common/common_behav.py

```diff
@@ -296,3 +296,165 @@ def get_abs_path(cls, key: Dict):
                 f"video file with filename: {video_filename} "
                 f"does not exist in {video_dir}/"
             )
+
+
+@schema
+class PositionIntervalMap(dj.Computed):
+    definition = """
+    -> IntervalList
+    ---
+    position_interval_name: varchar(200)  # name of the corresponding position interval
+    """
+
+    def make(self, key):
+        # Find correspondence between pos valid times names and epochs
+        # Use epsilon to tolerate small differences in epoch boundaries across epoch/pos intervals
+
+        # *** HARD CODED VALUES ***
+        epsilon = 0.11  # tolerated time difference in epoch boundaries across epoch/pos intervals
+        # *************************
+
+        # Unpack key
+        nwb_file_name = key["nwb_file_name"]
+
+        # Get pos interval list names
+        pos_interval_list_names = get_pos_interval_list_names(nwb_file_name)
+
+        # Skip populating if no pos interval list names
+        if len(pos_interval_list_names) == 0:
+            print(
+                f"NO POS INTERVALS FOR {key}; CANNOT POPULATE PositionIntervalMap"
+            )
+            return
+
+        # Get the interval times
+        valid_times = (IntervalList & key).fetch1("valid_times")
+        time_interval = [
+            valid_times[0][0] - epsilon,
+            valid_times[-1][-1] + epsilon,
+        ]  # [start, end], widen to tolerate small differences in epoch boundaries across epoch/pos intervals
+
+        # compare the position intervals against our interval
+        matching_pos_interval_list_names = []
+        for (
+            pos_interval_list_name
+        ) in pos_interval_list_names:  # for each pos valid time interval list
+            pos_valid_times = (
+                IntervalList
+                & {
+                    "nwb_file_name": nwb_file_name,
+                    "interval_list_name": pos_interval_list_name,
+                }
+            ).fetch1(
+                "valid_times"
+            )  # get interval valid times
+            pos_time_interval = [
+                pos_valid_times[0][0],
+                pos_valid_times[-1][-1],
+            ]  # [pos valid time interval start, pos valid time interval end]
+            if (time_interval[0] < pos_time_interval[0]) and (
```

Comment:
So the assumption here is that the position time interval is always contained within the interval (with some error). I think we have some other code just checking for any overlap in the intervals (in common interval) which might be more robust?

Reply:
Yeah looking for values from interval_list_intersect would be an easy way to check if there's any overlap.  I don't know if there's use cases where you would want it to return a value for partial overlap, or scenarios where there could be multiple small position intervals in an epoch though? This solution seemed the most strict but more likely to give 1:1 mapping

Reply:
There can be gaps in the position intervals due to loss of tracking but I agree that it probably won't be a problem